ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift

```diff
@@ -571,19 +574,76 @@ struct FrameworkBuilder {
     } catch {
       fatalError("Could not copy \(framework) to destination: \(error)")
     }
-    // Copy the module map to the destination.
-    let moduleDir = destination.appendingPathComponent("Modules")
-    do {
-      try FileManager.default.createDirectory(at: moduleDir, withIntermediateDirectories: true)
-    } catch {
-      fatalError("Could not create Modules directory for framework: \(framework). \(error)")
+
+    // For Swift modules, we use the modulemap from the xcodebuild. For Objective C modules,
+    // We use the constructed module map that includes required framework and library
+    // dependencies.
+    if !makeSwiftModuleMap(thinArchives: thinArchives, destination: destination) {
```

Comment:
Can you please break this out into separate lines with an appropriately named variable for the return statement? It's a little tough to read right now, reading it as `if don't make Swift module map` but the function itself has side-effects. Maybe something like:

```
let builtSwiftModules = makeSwiftModuleMap(thinArchives: thinArchives, destination: destination)
if !builtSwiftModules {
  // Contents of block...
}
```